### PR TITLE
[nn] Add Triton-accelerated RMSNorm implementation (3.58x speedup)

### PR DESCRIPTION
**Pull request by JonSnow1807**
## Summary

This PR adds an optional Triton-based implementation of RMSNorm that provides significant speedups for memory-bound scenarios.

## Motivation

RMSNorm is becoming increasingly popular in modern transformer architectures (LLaMA, Mistral, etc.). The current PyTorch implementation, while correct, can be a bottleneck in memory-bound scenarios. This PR provides a fast alternative using Triton kernels.

## Implementation

- Adds `FastRMSNorm` class that uses optimized Triton kernels
- Provides `enable_fast_rmsnorm()` / `disable_fast_rmsnorm()` for global opt-in
- Falls back to standard implementation on CPU or when Triton is unavailable
- Fully compatible with existing code

## Performance

Benchmarks on Tesla T4:
- Standalone RMSNorm: **3.58x faster**
- Multi-layer normalization: **2.01x faster**  
- Attention patterns: **~1.5x faster**

## Testing

- Correctness tests against standard implementation (max error: 9.5e-7)
- Performance benchmarks
- CPU fallback tests
- Global enable/disable tests

## Usage

```python
# Option 1: Global enable
torch.nn.enable_fast_rmsnorm()
model = TransformerModel()  # All RMSNorm layers will be fast

# Option 2: Direct usage
from torch.nn import FastRMSNorm
norm = FastRMSNorm(hidden_size)
```

## Test Results

```
Correctness: max diff = 9.5e-07 (PASS)
Performance: 0.1487s vs 0.0416s = 3.58x speedup
```

## Implementation Details

The Triton kernel optimizes memory access patterns and reduces kernel launch overhead:

```python
@triton.jit
def rmsnorm_kernel(X, W, Y, stride_xm, stride_xn, N, eps, BLOCK_SIZE):
    # Efficient row-wise computation
    # Single pass for RMS calculation
    # Fused normalization and scaling
```

Key optimizations:
- Vectorized memory loads
- Reduced memory traffic (single pass)
- Optimal block sizes for different hardware
- Automatic fallback for edge cases

## Compatibility

- PyTorch 2.0+ (uses Triton when available)
- CUDA compute capability 7.0+ 
- Tested on T4, V100, A100 GPUs
- CPU fallback ensures universal compatibility

## Future Work

- Extend to 2D/3D normalization shapes
- Add FP16/BF16 optimized paths
- Integrate with torch.compile for automatic pattern matching

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

### CODE REVIEW

**Comment by vadimkantorov** [comment]
There was also recently a merged PR with fused impl of RMSNorm:
- https://github.com/pytorch/pytorch/pull/153666
